lib/queuePopulator/LogReader.js

```diff
@@ -330,15 +318,29 @@ class LogReader {
             }
             return key;
         }
-        if (entry.key === undefined || entry.value === undefined) {
-            return;
+        // TODO: handle this better, this is too specific to notifications
```

Comment:
:+1: 

If you have time let's just fix it now, I think it's only a matter of doing the same special processing in the filter function of BN itself correct?

Reply:
Ah I think I see what you mean, because in the next PR you intend to add the `cb` arg to the filter function for BN? That's fine for me, but don't refrain from being more specific in the TODO then.